CRM/Core/Page/AJAX/Location.php

```diff
@@ -180,8 +180,9 @@ public static function getPermissionedLocation() {
           }
           elseif ($htmlType == 'Select Date') {
             $elements["onbehalf_{$key}"]['type'] = $htmlType;
-            $elements["onbehalf_{$key}"]['value'] = $defaults[$key];
-            $elements["onbehalf_{$key}_display"]['value'] = $defaults[$key];
+            //CRM-18349, date value must be ISO formatted before being set as a default value for crmDatepicker custom field
+            $elements["onbehalf_{$key}"]['value'] = CRM_Utils_Date::processDate($defaults[$key], NULL, FALSE, 'Y-m-d G:i:s');
+            $elements["onbehalf_{$key}_display"]['value'] = CRM_Utils_Date::processDate($defaults[$key], NULL, FALSE, 'Y-m-d G:i:s');
```

Comment:
The second element `"onbehalf_{$key}_display"` should no longer exist (the weirdness btw `field` and `field_display` has been fixed). Try removing this second line and it will probably still work fine.